### ISSUE TITLE
Duplicated output upon rebuild 

### ISSUE THREAD

**Issue by hidmic** [ros-infrastructure/rosdoc2#32]
Rebuilding a package's documentation with `rosdoc2 build` using the same build and cross-reference directories induces duplication warnings. These warnings go away after nuking said directories. 

Should `rosdoc2` do some housekeeping _before_ generating output?

**Comment by aprotyas** [comment]
Housekeeping in the sense that if `rosdoc2 build --package-path P --build-directory B --cross-reference-directory C` were invoked, we'd want to prune `B/P` and `C/P` given they exist, right? If so, that sounds reasonable.

**Comment by hidmic** [comment]
@aprotyas that'd be an option. I haven't dug enough to know how much can be reused.

**Comment by aprotyas** [comment]
Yes, it would be nice to re-use some things. IIRC, the doxygen artifacts (XML tags, etc.) don't get duplicated.
